src/hooks/useProjects/useProjects.ts

```diff
@@ -0,0 +1,36 @@
+import { useCallback, useEffect, useState } from "react";
+import { database } from "services";
+import { IProject } from "types";
+
+export const useProjects = () => {
+  const [projects, setProjects] = useState<IProject[] | null>(null);
+  const [project, setProject] = useState<IProject | null>(null);
+
+  const getProjects = useCallback(
+    () => database.getAllProjects(setProjects),
+    []
+  );
+
+  const getProject = useCallback(
+    (slug: string) => database.getProject(slug, setProject),
+    []
+  );
+
+  const updateProject = useCallback(
+    (project: IProject) => {
+      database.updateProject(project);
+      getProject(project.slug);
+    },
+    [getProject]
+  );
+
+  const resetProject = useCallback(() => setProject(null), []);
```

Comment:
This may not be necessary because it looks like `project` gets reset to null automatically on unmount.